Libjxl static build requires -DJPEGXL_ENABLE_SKCMS=false to link

**Describe the bug**

Unless -DJPEGXL_ENABLE_SKCMS=false is specified in a static build, a dependent application is not able to link due to unresolved skcms symbols, which appear to be in an uninstalled libskcms.a.  This includes when -DJPEGXL_BUNDLE_SKCMS=true.

**To Reproduce**
Steps to reproduce the behavior:

With libraries (libpng, libjpeg, liblcms2) built static in advance, this build recipe is used:

```
    cmake \
        -DCMAKE_INSTALL_PREFIX=$WORK \
        -DBUILD_TESTING=off \
        -DBUILD_SHARED_LIBS=false \
        -DCMAKE_BUILD_TYPE=Release \
        -DJPEGXL_ENABLE_BENCHMARK=false \
        -DJPEGXL_ENABLE_EXAMPLES=false \
        -DJPEGXL_ENABLE_FUZZERS=false \
        -DJPEGXL_ENABLE_JPEGLI=false \
        -DJPEGXL_ENABLE_MANPAGES=OFF \
        -DJPEGXL_ENABLE_SJPEG=false \
        -DJPEGXL_ENABLE_TOOLS=false \
        -DJPEGXL_ENABLE_VIEWERS=false \
        -DJPEGXL_ENABLE_JPEGLI=false \
        -DJPEGXL_FORCE_SYSTEM_LCMS2=false \
        -DCMAKE_C_FLAGS="-DHWY_DISABLED_TARGETS=HWY_SSSE3 ${CFLAGS} -fPIC" \
        -DCMAKE_CXX_FLAGS="-DHWY_DISABLED_TARGETS=HWY_SSSE3 ${CXXFLAGS} -fPIC" \
        "${SRC}/libjxl"
```
With the above, a spew of unresolved symbols are produced when performing a static link with a dependent application (GraphicsMagick) for purpose of oss-fuzz testing.  However, this recipe avoids the unresolved symbols:

```
    cmake \
        -DCMAKE_INSTALL_PREFIX=$WORK \
        -DBUILD_TESTING=off \
        -DBUILD_SHARED_LIBS=false \
        -DCMAKE_BUILD_TYPE=Release \
        -DJPEGXL_ENABLE_BENCHMARK=false \
        -DJPEGXL_ENABLE_EXAMPLES=false \
        -DJPEGXL_ENABLE_FUZZERS=false \
        -DJPEGXL_ENABLE_JPEGLI=false \
        -DJPEGXL_ENABLE_MANPAGES=OFF \
        -DJPEGXL_ENABLE_SJPEG=false \
        -DJPEGXL_ENABLE_TOOLS=false \
        -DJPEGXL_ENABLE_VIEWERS=false \
        -DJPEGXL_ENABLE_JPEGLI=false \
        -DJPEGXL_ENABLE_SKCMS=false \
        -DJPEGXL_BUNDLE_SKCMS=false \
        -DJPEGXL_FORCE_SYSTEM_LCMS2=false \
        -DCMAKE_C_FLAGS="-DHWY_DISABLED_TARGETS=HWY_SSSE3 ${CFLAGS} -fPIC" \
        -DCMAKE_CXX_FLAGS="-DHWY_DISABLED_TARGETS=HWY_SSSE3 ${CXXFLAGS} -fPIC" \
        "${SRC}/libjxl"
```

**Expected behavior**

It is expected that when a dependent application links using the linker flags output by 'pkg-config --static libjxl --libs' that linking will be successful.  But this is not possible since the skcms symbols are not included in libjxl.a.


**Environment**
 - OS: Linux
 - Compiler version: clang version 15.0.0
 - CPU type: x86_64
 - cjxl/djxl version string: cjxl v0.10.0 733dfda [AVX3,AVX2,SSE4,SSE2]

**Additional context**
None

For me, it fails to link no matter if it's using lcms or skcms.
It just fails with different missing symbols
```
...
jxl_cms.cc:(.text+0x33a8): undefined reference to `cmsCloseProfile'
...
```
and so on.

As it stands right now, statically linking latest libjxl with build-in l/skcms is impossible.

--- a/lib/jxl/cms/jxl_cms.cc
+++ b/lib/jxl/cms/jxl_cms.cc
@@ -1324,7 +1324,7 @@ float* JxlCmsGetDstBuf(void* cms_data, size_t thread) {
 
 extern "C" {
 
-const JxlCmsInterface* JxlGetDefaultCms() {
+JXL_CMS_EXPORT const JxlCmsInterface* JxlGetDefaultCms() {
   static constexpr JxlCmsInterface kInterface = {
       /*set_fields_data=*/nullptr,
       /*set_fields_from_icc=*/&JxlCmsSetFieldsFromICC,

--- a/lib/jxl_cms.cmake
+++ b/lib/jxl_cms.cmake
@@ -12,7 +12,10 @@ add_library(jxl_cms
   ${JPEGXL_INTERNAL_CMS_SOURCES}
 )
 target_compile_options(jxl_cms PRIVATE "${JPEGXL_INTERNAL_FLAGS}")
-set_target_properties(jxl_cms PROPERTIES POSITION_INDEPENDENT_CODE ON)
+set_target_properties(jxl_cms PROPERTIES
+        POSITION_INDEPENDENT_CODE ON
+        CXX_VISIBILITY_PRESET hidden
+        VISIBILITY_INLINES_HIDDEN 1)
 target_link_libraries(jxl_cms PUBLIC jxl_base)
 target_include_directories(jxl_cms PRIVATE
   ${JXL_HWY_INCLUDE_DIRS}
@@ -26,7 +29,7 @@ target_include_directories(jxl_cms PUBLIC
 set(JPEGXL_CMS_LIBRARY_REQUIRES "")
 
 if (JPEGXL_ENABLE_SKCMS)
-  target_link_libraries(jxl_cms PRIVATE skcms)
+  target_link_skcms(jxl_cms)
 else()
   target_link_libraries(jxl_cms PRIVATE lcms2)
   if (JPEGXL_FORCE_SYSTEM_LCMS2)

